lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -322,6 +323,28 @@ private void checkStorageCode(final String storageCode) {
         };
     }
 
+    public Map<String, String> getWorkspaceInfo(
+            final long wsId)
+            throws RetriableIndexingException, IndexingException {
+
+        final Map<String, Object> command = new HashMap<>();
+        command.put("command", "getWorkspaceInfo");
+        command.put("params", new WorkspaceIdentity()
+                .withId(wsId));
+
+        final Map<String, String> wsInfoMeta;
+
+        try {
+            wsInfoMeta = ws.getClient().administer(new UObject(command))
+                         .asClassInstance(WS_INFO_TYPEREF).getE9();
+        } catch (IOException e) {
+            throw handleException(e);
```

Comment:
I'm not totally sure that this method makes sense for this class, but I suppose it doesn't hurt. The exception handlers here don't really make a ton of sense for the search api use case because the wrap the IO and JC exceptions in IndexingException, which isn't appropriate for the API because it doesn't have the machinery to respond appropriately to those exceptions.

Reply:
Removed the wrappers for exceptions.